### PR TITLE
Don't warn about equal timestamps during append.

### PR DESCRIPTION
**Pull request by laushinka**
Perhaps it would be even better to still warn in case the sample value has
changed but the timestamps are equal, but we don't have fast access to the
last value.

(paired on this with @juliusv)

### CODE REVIEW

**Comment by fabxc** [comment]
Thanks!
Can you give some context about where appending of the exact timestamp twice came up?

**Comment by brian-brazil** [comment]
This will come up once https://github.com/prometheus/prometheus/issues/398 is resolved, as then the pushgateway will be exposing timestamps.

**Comment by laushinka** [comment]
Yes, client-side timestamps (especially in the pushgateway) will be a common use case that will trigger equal timestamps.

File: storage/local/storage.go [hunk 1]
```diff
@@ -560,8 +560,10 @@ func (s *memorySeriesStorage) Append(sample *clientmodel.Sample) {
 	series := s.getOrCreateSeries(fp, sample.Metric)
 
 	if sample.Timestamp <= series.lastTime {
+		if sample.Timestamp != series.lastTime {
+			log.Warnf("Ignoring sample with out-of-order timestamp for fingerprint %v (%v): %v is not after %v", fp, series.metric, sample.Timestamp, series.lastTime)
+		}
 		s.fpLocker.Unlock(fp)
-		log.Warnf("Ignoring sample with out-of-order timestamp for fingerprint %v (%v): %v is not after %v", fp, series.metric, sample.Timestamp, series.lastTime)
 		s.outOfOrderSamplesCount.Inc()
```
Comment (brian-brazil) [inline storage/local/storage.go]: We probably only want to increment this if the timestamp is different

Comment (laushinka) [inline storage/local/storage.go]: Yes, after reconsidering, this probably makes the most sense. Updated.

**Comment by fabxc** [comment]
Thanks for the clarification. So this will also come up with federation if we scrape `/federate` more frequently than the values change.

Can we add a comment to clarify this as it's not self-explanatory without the value comparison.
After that :+1:

**Comment by laushinka** [comment]
@fabxc Good point. Added a comment.

**Comment by fabxc** [comment]
Great, thanks! :+1: :ship:

**Pull request merged**